### PR TITLE
remove licenses of removed libraries, update the license checker 

### PR DESCRIPTION
**Pull request by janjwerner-confluent**
### Description
- Licenses file contains several licenses for outdated libraries. In this PR we remove licenses for no longer used components. 
This change is purely cosmetic / cleans up the license database. 
The candidates were designated by reviewing the output of the license check script and comparing it against the depdency tree. 
 - Minor fix to license check tool to fail more gracefully when the license of used dependency is not listed as known, as well as fix not to fail on multi licensed components when at least one of the licenses is accepted. 
 
This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: distribution/bin/check-licenses.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import sys
 from html.parser import HTMLParser
 import argparse
+#import json
```
Comment (xvrl) [inline distribution/bin/check-licenses.py:23]: let's remove this line if it is not necessary

**Review by xvrl — APPROVED** [review]
minor nit, otherwise LGTM

File: distribution/bin/check-licenses.py (hunk 1 above)
Comment (xvrl) [inline distribution/bin/check-licenses.py:23]:
```suggestion
```

Comment (janjwerner-confluent) [reply distribution/bin/check-licenses.py]: done!

**Pull request merged**